HelloPush/app/src/main/java/org/jboss/aerogear/android/cookbook/hellopush/handler/NotificationBarMessageHandler.java

```diff
@@ -49,23 +65,54 @@ public void onMessage(Context context, Bundle bundle) {
         notify(context, bundle);
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    private void createChannel() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            throw new IllegalStateException("This function should not be called on < Android O");
+        }
+
+        NotificationManager notificationManager =
+                (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+
+        // The user-visible name of the channel.
+        CharSequence name = context.getString(R.string.channel_name);
+
+        // The user-visible description of the channel.
+        String description = context.getString(R.string.channel_description);
+        int importance = NotificationManager.IMPORTANCE_LOW;
+        NotificationChannel channel = new NotificationChannel(CHANNEL_ID, name, importance);
+
+        // Configure the notification channel.
+        channel.setDescription(description);
+        channel.enableLights(true);
+
+        // Sets the notification light color for notifications posted to this
+        // channel, if the device supports this feature.
+        channel.setLightColor(Color.RED);
+        channel.enableVibration(true);
+        channel.setVibrationPattern(new long[]{100, 200, 300, 400, 500, 400, 300, 200, 400});
+        notificationManager.createNotificationChannel(channel);
+
+    }
+
     private void notify(Context context, Bundle bundle) {
         NotificationManager mNotificationManager = (NotificationManager)
                 context.getSystemService(Context.NOTIFICATION_SERVICE);
 
         String message = bundle.getString(UnifiedPushMessage.ALERT_KEY);
         String pushMessageId = bundle.getString(UnifiedPushMessage.PUSH_MESSAGE_ID);
 
+        @SuppressLint("WrongConstant")
         Intent intent = new Intent(context, MessagesActivity.class)
                 .addFlags(PendingIntent.FLAG_UPDATE_CURRENT)
```

Comment:
@secondsun Android Studio doesn't seem to like `PendingIntent.FLAG_UPDATE_CURRENT` here - might get your thoughts, or @danielpassos thoughts on it, if you open the code at some point